modules/Oculus_module/src/OculusModule.cpp

```diff
@@ -416,6 +418,73 @@ bool OculusModule::configure(yarp::os::ResourceFinder& rf)
     }
     m_oculusHeadsetPoseInertial.resize(6, 0.0);
 
+    //Reset the cameras if necessary
+    bool resetCameras = generalOptions.check("resetCameras", yarp::os::Value(false)).asBool();
+    yInfo() << "Reset camera: " << resetCameras;
+    if (resetCameras)
+    {
+        auto cameraReset = [](const std::string& cameraPort, const std::string& localPort) -> bool
+        {
+            yarp::dev::PolyDriver grabberDriver;
+
+            yarp::os::Property config;
+            config.put("device", "remote_grabber");
+            config.put("remote", cameraPort);
+            config.put("local", localPort);
+
+            bool opened = grabberDriver.open(config);
+            if(!opened)
+            {
+                yError() << "Cannot open remote_grabber device on port " << cameraPort <<".";
+                return false;
+            }
+
+            yarp::dev::IFrameGrabberControlsDC1394 *grabberInterface;
+
+            if(!grabberDriver.view(grabberInterface) || !grabberInterface)
+            {
+                yWarning() << "RemoteGrabber does not have IFrameGrabberControlDC1394 interface, please update yarp.";
+                return false;
+            }
+
+            if(!grabberInterface->setResetDC1394())
+            {
+                yWarning() << "Failed to reset the camera on port " << cameraPort << ".";
+                return false;
+            }
+
+            grabberDriver.close();
+
+            return true;
+        };
+
+        std::string leftCamera, rightCamera;
+        if (!YarpHelper::getStringFromSearchable(generalOptions, "leftCameraPort", leftCamera))
+        {
+            yError() << "[OculusModule::configure] resetCameras is non-zero, but no leftCameraPort is provided.";
+            return false;
+        }
+
+        if (!YarpHelper::getStringFromSearchable(generalOptions, "rightCameraPort", rightCamera))
+        {
+            yError() << "[OculusModule::configure] resetCameras is non-zero, but no rightCameraPort is provided.";
+            return false;
+        }
+
+        if (!cameraReset(leftCamera,"/walking-teleoperation/camera-reset/left"))
+        {
+            yError() << "Failed to reset left camera.";
```

Comment:
```suggestion
            yError() << "[OculusModule::configure] Failed to reset left camera.";
```

Reply:
also for the other prints

Reply:
Good point. I added it in https://github.com/robotology/walking-teleoperation/pull/45/commits/4f76580403f63782e5611cdc0326a2f2649b33c5